cmake/list_insert_options.cmake

```diff
@@ -0,0 +1,55 @@
+#
+# Append key[=value] elements from one list to another list if they are not already in the target list.
```

Comment:
Right now I don't know if we should merge it.
It adds complexity, but might lead to problems if it gets used widely.

For the C++ standards issue I have implemented [other work-arounds](https://github.com/ipa-mdl/enable_cxx11_support).